SpatialGDK/Source/SpatialGDK/Public/LoadBalancing/ReferenceCountedLockingPolicy.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Containers/Map.h"
+#include "Containers/UnrealString.h"
+#include "GameFramework/Actor.h"
+
+#include "AbstractLockingPolicy.h"
+#include "ReferenceCountedLockingPolicy.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogReferenceCountedLockingPolicy, Log, All)
+
+UCLASS()
+class SPATIALGDK_API UReferenceCountedLockingPolicy : public UAbstractLockingPolicy
+{
+	GENERATED_BODY()
+
+public:
+	virtual ActorLockToken AcquireLock(const AActor* Actor) override;
+	virtual ActorLockToken AcquireLock(const AActor* Actor, FString DebugString) override;
+
+	virtual void ReleaseLock(ActorLockToken Token) override;
```

Comment:
As I understand, the interface has changed, and the locks are not released, when leaving the scope anymore? Is it something we intend to revisit in the future? Could you mention this design change in description?

Reply:
The main reason is that you can't create compositional policies using a scoped thing but you can with tokens. You can make an RAII token on top of this though by taking the top level policy and putting it into a unique_ptr or similar. We just haven't made that user API on top yet.
I wanted to see if it had any ergonic issues too as it might be that we want that idea but not in quite the same way.

Reply:
It can be made on top of this though.

Reply:
Resolving, since we've decided to not act on this one for now.